internal/core/operations/remove_rooms/remove_rooms_executor.go

```diff
@@ -111,35 +116,36 @@ func (e *RemoveRoomsExecutor) removeRoomsByIDs(ctx context.Context, schedulerNam
 		rooms = append(rooms, gameRoom)
 	}
 
-	err := e.deleteRooms(ctx, rooms)
+	err := e.deleteRooms(ctx, logger, rooms)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (e *RemoveRoomsExecutor) removeRoomsByAmount(ctx context.Context, schedulerName string, amount int) error {
+func (e *RemoveRoomsExecutor) removeRoomsByAmount(ctx context.Context, logger *zap.Logger, schedulerName string, amount int) error {
 	rooms, err := e.roomManager.ListRoomsWithDeletionPriority(ctx, schedulerName, "", amount, &sync.Map{})
 	if err != nil {
 		return err
 	}
 
-	err = e.deleteRooms(ctx, rooms)
+	err = e.deleteRooms(ctx, logger, rooms)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (e *RemoveRoomsExecutor) deleteRooms(ctx context.Context, rooms []*game_room.GameRoom) error {
+func (e *RemoveRoomsExecutor) deleteRooms(ctx context.Context, logger *zap.Logger, rooms []*game_room.GameRoom) error {
 	var err error
 	for _, room := range rooms {
 		err = e.roomManager.DeleteRoomAndWaitForRoomTerminated(ctx, room)
```

Comment:
Shouldn't we implement the `forwardingStatusTerminatingEvent` inside the `DeleteRoomAndWaitForRoomTerminated`? I think it'll be better since we should re-implement it on the `switch active version` operation too, creating 2 dependencies (events service) and code duplication